cli/tools/installer.rs

```diff
@@ -152,12 +152,19 @@ pub async fn infer_name_from_url(url: &Url) -> Option<String> {
     return None;
   }
 
-  let path = PathBuf::from(url.path());
-  let mut stem = match path.file_stem() {
-    Some(stem) => stem.to_string_lossy().to_string(),
-    None => return None,
+  let percent_decode = percent_encoding::percent_decode(url.path().as_bytes());
+  #[cfg(unix)]
+  let path = {
+    use std::os::unix::prelude::OsStrExt;
+    PathBuf::from(std::ffi::OsStr::from_bytes(
+      &percent_decode.collect::<Vec<u8>>(),
+    ))
```

Comment:
```suggestion
    use std::os::unix::prelude::OsStringExt;
    PathBuf::from(std::ffi::OsString::from_vec(
      percent_decode.collect::<Vec<u8>>(),
    ))
```

> This conversion does not allocate or copy memory.

https://doc.rust-lang.org/std/path/struct.PathBuf.html#impl-From%3COsString%3E-for-PathBuf